OpenUpSA/wazimap-ng-ui
# Bug/fixed map default coordinates

## Description

Fixed default coordinates in config

Default coordinates can be set in the profile config. These will set the initial lat/lon/zoom when no geography is set in the hash fragment, i.e. /#geo:<geography> - previously this didn't not work as the default coordinates were set and then the map immediately panned to the default geography. This code keeps the shouldZoom state in the controller. shouldZoom is disabled when there is no geography in the fragment.

No tests were written because the jest complains about leaflet and I could not get `transformIgnorePatterns` to work

## Related Issue
N/A

## How to test it locally
In a profile config set default coordinates, e.g. 

```
{
  ...
  "default_coordinates": {
    "lat": 10,
    "long": 15,
    "zoom": 3
  }
}
```

When you navigate to the root url the view should be set appropriately. You should still be able to click on other geographies after which zoom and pan works as normal.

## Screenshots


## Changelog

### Added

### Updated
controller.js now stores shouldZoom state. maps.js will then check that state when deciding whether to zoom on click or not

### Removed


## Checklist


### Pull Request


### Commits


### Code Quality


### Testing
